IdentityCore/src/webview/systemWebview/session/MSIDSystemWebViewControllerFactory.m

```diff
@@ -39,25 +39,9 @@ + (MSIDWebviewType)availableWebViewTypeWithPreferredType:(MSIDWebviewType)prefer
     {
         return preferredType;
     }
-    
-#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 120000 || __MAC_OS_X_VERSION_MAX_ALLOWED >= 101500
-        
-    if (@available(iOS 12.0, macOS 10.15, *))
-    {
-        return MSIDWebviewTypeAuthenticationSession;
-    }
-#endif
-        
-#if TARGET_OS_IPHONE && !TARGET_OS_MACCATALYST
-        
+#if (TARGET_OS_IPHONE || TARGET_OS_MAC) && !TARGET_OS_MACCATALYST
```

Comment:
There seems to be one more condition that old code was taking care of that is when target is iPhone alone . I think we should add it back

#if TARGET_OS_IPHONE
    return MSIDWebviewTypeSafariViewController;
#endif

Reply:
On a second look, the previous code was returning MSIDWebviewTypeAuthenticationSession if target iphone is >=12 or macOS >= 10.15. Now that both target conditions are satisfying, we only need to return MSIDWebviewTypeAuthenticationSession from this api 

+ (MSIDWebviewType)availableWebViewTypeWithPreferredType:(MSIDWebviewType)preferredType
{
    if (preferredType != MSIDWebviewTypeDefault)
    {
        return preferredType;
    }
    
    return MSIDWebviewTypeAuthenticationSession;
    
}

Reply:
Indeed, this condition
```
#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 120000 || __MAC_OS_X_VERSION_MAX_ALLOWED >= 101500
        
    if (@available(iOS 12.0, macOS 10.15, *))
    {
        return MSIDWebviewTypeAuthenticationSession;
    }
#endif
```
would take precedence over 
```
#if TARGET_OS_IPHONE && !TARGET_OS_MACCATALYST
        
    return MSIDWebviewTypeAuthenticationSession;
        
#endif
```
Because based on 2-nd condition 
`TARGET_OS_IPHONE && TARGET_OS_MACCATALYST`
is same as
`TARGET_OS_IPHONE`
As in an `TARGET_OS_IPHONE` is also a `TARGET_OS_MACCATALYST but not all `TARGET_OS_MACCATALYST` targets are  `TARGET_OS_IPHONE`.
Otherwise there is no way the 2nd would ever validate. 
Updated.